manifest-build-tools/application/update_changelog.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, EMC, Inc.
+
+"""
+The script update changelog under debian
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/update_changelog.py \
+--build-dir d/ \
+--version 1.2.6 \
+--publish \
+--git-credential https://github.com,GITHUB \
+--message "new branch 1.2.6"
+
+The required parameters: 
+build-dir: The top directory which stores all the cloned repositories
+version: The new release version
+
+The optional parameters:
+message (default value is "new release" + version )
+publish: If true, the updated changlog will be push to github.
+git-credential: url, credentials pair for the access to github repos.
+                For example: https://github.com,GITHUB
+                GITHUB is an environment variable: GITHUB=username:password
+                If parameter publish is true, the parameter is required.
+"""
+import os
+import sys
+import argparse
+import datetime
+import subprocess
+from RepositoryOperator import RepoOperator
+from common import *
+
+class ChangelogUpdater(object):
+    def __init__(self, repo_dir, version):
+        """
+        The module updates debian/changelog under the directory of repository
+        _repo_dir: the directory of the repository
+        _version: the new version which is going to be updated to changelog
+        """
+        self._repo_dir = repo_dir
+        self._version = version
+        self.repo_operator = RepoOperator()
+ 
+    def debian_exist(self):
+        """
+        check whether debian directory under the repository
+        return: True if debian exist
+                False
+        """
+        if os.path.isdir(self._repo_dir):
+            for filename in os.listdir(self._repo_dir):
+                if filename == "debian":
+                    return True
+        return False
+
+    def get_repo_name(self):
+        """
+        get the name of the repository
+        :return: the name of the repository
+        """
+        repo_url = self.repo_operator.get_repo_url(self._repo_dir)
+        repo_name = strip_suffix(os.path.basename(repo_url), ".git")
+        return repo_name
+
+    def update_changelog(self, message=None):
+        """
+        add an entry to changelog
+        :param message: the message which is going to be added to changelog
+        return: Ture if changelog is updated
+                False, otherwise
+        """
+        repo_name = self.get_repo_name()
+        debian_exist = self.debian_exist()
+        linked = False
+        if not debian_exist:
+            for filename in os.listdir(self._repo_dir):
+                if filename == "debianstatic":
+                    debianstatic_dir = os.path.join(self._repo_dir, "debianstatic")
+                    for debianstatic_filename in os.listdir(debianstatic_dir):
+                        if debianstatic_filename == repo_name:
```

Comment:
It's a special handling for on-http repo which contains debianstatic/on-http folder (as a "debian" folder template).
May I suggest you to write some comments in code ( because code here is for above special cases)